### PR TITLE
fix(web-components): remove aria-hidden in see more/less buttons

### LINKED ISSUE #1854: [ic-typography] accessibility error on see more/see less button
## Summary of the bug 
When running accessibility tests using accessibility insights and the storybook a11y add on, the same error occurs. 

aria-hidden-focus - ensure aria-hidden components aren’t focusable. 

This is because the see more/see less typography button is focusable and aria-hidden

## 🪜 How to reproduce 
Tell us the steps to reproduce the problem: 
1. Go to truncation example on storybook
2. Click on the accessibility tab in storybook or run an accessibility insights fast pass
3. See the error 

## 🧐 Expected behaviour 
It shouldn't produce this accessibility error.

### PR DESCRIPTION
**Pull request by evenstensberg**
## Summary of the changes
Removes aria hidden attribute and enables skipped a11y tests in ic -typography.

## Related issue
Fixes #1854

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.

### CODE REVIEW

**Comment by GCHQ-Developer-530** [comment]
Hi @evenstensberg, part of #1854 is to uncomment out the a11y checks in the cypress tests for typography? Please could you add a new commit for that work? Thank you

File: packages/web-components/src/components/ic-typography/ic-typography.tsx [hunk 1]
```diff
@@ -203,7 +203,7 @@ export class Typography {
             onBlur={this.truncButtonBlur}
             onMouseDown={this.truncButtonFocusFromMouse}
             onClick={this.toggleExpanded}
-            aria-hidden="true"
+            tabindex="0"
```
Comment (ad9242) [inline packages/web-components/src/components/ic-typography/ic-typography.tsx:206]: I don't think that explicitly setting `tabindex` to `0` is required, as a `<button>` will be focusable by default. So hopefully just removing aria-hidden is all that is required. The unit tests will fail as a result, so the snapshot files should be updated to reflect the change

Comment (evenstensberg) [reply packages/web-components/src/components/ic-typography/ic-typography.tsx]: @ad9242 which npm command updates snapshots, and snapshots only?

**Comment by evenstensberg** [comment]
@GCHQ-Developer-530 Rebased the change, but need to update the snapshots too.

File: packages/web-components/src/components/ic-typography/ic-typography.tsx (hunk 1 above)
Comment (ad9242) [reply packages/web-components/src/components/ic-typography/ic-typography.tsx]: I don't think there is an npm command for it, but you can delete the existing snapshot file & then run the tests locally (npm run test), to have it regenerate the file and then validate manually that the changes are what you would expect

**Comment by evenstensberg** [comment]
@ad9242 @GCHQ-Developer-530 snapshot is updated, PTAL!

**Comment by GCHQ-Developer-530** [comment]
Hi @evenstensberg, this work still needs the cypress update I mentioned in my previous comment. It should just be a case of uncommenting out any checkA11yWithWait() commands from the typography tests e.g. [this line in the file](https://github.com/mi6/ic-ui-kit/blob/develop/packages/react/src/component-tests/IcTypography/IcTypography.cy.tsx#L135) should be uncommented. Thanks

**Comment by GCHQ-Developer-530** [comment]
Also, please can you squash your 2 web-components scoped commits into one commit. Thank you

**Comment by evenstensberg** [comment]
@GCHQ-Developer-530 Done. As the a11y test was in react package it's in its own commit.

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Review by ad9242 — APPROVED** [review]

**Pull request merged**